### PR TITLE
Add auto complete name to class, interface, enum, struct etc.

### PR DESCRIPTION
**Pull request by MRmlik12**
Add auto completion for class, interface, enum, struct to improve coding productivity

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=5198) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=5198) before we can accept your contribution.<br/><br/>:x: MRmlik12 [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=5198)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-vscode?pullRequest=5198) it.</sub>

File: snippets/csharp.json [hunk 1]
```diff
@@ -198,7 +198,7 @@
     "Interface": {
         "prefix": "interface",
         "body": [
-            "interface I${1:Name}",
+            "interface I${1:$TM_FILENAME_BASE}",
```
Comment (JoeRobich) [inline snippets/csharp.json:201]: Do we expect the filename would already included the I prefix?

```suggestion
            "interface ${1:$TM_FILENAME_BASE}",
```

Comment (MRmlik12) [reply snippets/csharp.json:201]: Oh, I must remove by accident :D

Comment (MRmlik12) [reply snippets/csharp.json:201]: In default, every c# project has `I` in filename for example https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/ICloneable.cs

Comment (JoeRobich) [reply snippets/csharp.json:201]: Would this template generate `interface IICloneable` for that file?

Comment (MRmlik12) [reply snippets/csharp.json]: My bad, I fixed this

**Review by JoeRobich — APPROVED** [review]
Thanks @MRmlik12!

**Comment by LukeWolfGames** [comment]
Is there a way to turn this off in the extention settings because this seems to interfere with coding in Unity due to some of Unity's classes?

**Comment by JoeRobich** [comment]
@LukeWolfGames These changes were for snippets and not for completion suggestions. Could you open a new issue to describe the trouble you are running into?

**Pull request merged**